code/script.js

```diff
@@ -225,101 +232,164 @@ const generateBoard = () => {
+      alert(`No, the person doesnt have ${value} eyes! Remove all people with ${value} eyes`)
+      charactersInPlay = charactersInPlay.filter((person) => person['eyes'] !== value)
     }
   }
+  //if (category === 'hair')
+  else {
+    if (keep){
+      alert(`Yes, the person has ${value} hair! Keep all people with ${value} hair`)
+      charactersInPlay = charactersInPlay.filter((person) => person['hair'] === value)
+    } else {
+      alert (`No, the person doesnt have ${value} hair! Remove all people with ${value} hair`)
+      charactersInPlay = charactersInPlay.filter((person) => person['hair'] !== value)
+    }
+   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
-      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
-      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
-
-  // Invoke a function to redraw the board with the remaining people.
+   // Invoke a function to redraw the board with the remaining people.
+   generateBoard()
 }
 
 // when clicking guess, the player first have to confirm that they want to make a guess.
 const guess = (personToConfirm) => {
   // store the interaction from the player in a variable.
   // remember the confirm() ?
+  let playerGuess = confirm (`Do you want to make a guess on ${personToConfirm}?`)
   // If the player wants to guess, invoke the checkMyGuess function.
+  if (playerGuess === true){
+    checkMyGuess(personToConfirm)
+    console.log('a guess is made')
+  }else {
+    alert('Okey, lets keep guessing')
+    console.log('No guess made, keep playing')
+  }
 }
 
 // If you confirm, this function is invoked
-const checkMyGuess = (personToCheck) => {
+const checkMyGuess = (personToConfirm) => {
   // 1. Check if the personToCheck is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
+  if (personToConfirm === secret.name) {
+    // 2. Set a Message to show in the win or lose section accordingly
+    winOrLoseText.innerHTML = `Great job! You guessed on ${personToConfirm} an it was the right answer.`
+  } else {
+    winOrLoseText.innerHTML = `Better luck next time! You guessed on ${personToConfirm} but the correct answer is ${secret.name}.`
+  }
   // 3. Show the win or lose section
+  winOrLose.style.display = 'flex'
   // 4. Hide the game board
+  board.style.display = 'none'
+  
 }
 
 // Invokes the start function when website is loaded
 start()
 
+//restart game
+const newGame = () =>{
+winOrLose.style.display = 'none'
+board.style.display = 'flex'
+start()
+}
```

Comment:
As mentioned up in the start function, this newGame function is sort of repetitive. You could instead move two lines into the start function and always call start weather you load the page, restart the game or choose to play again after making a guess.